### PR TITLE
Get rid of `temp` package

### PR DESCRIPTION
**Pull request by juanarbol**
The `temp` package is not required for what it does, I would suggest using Node.js API to achieve the same thing.

### CODE REVIEW

**Comment by partouf** [comment]
Unfortunately we very much rely on the periodic cleanup in our live setup, so unless you can reintroduce that we cannot merge this.

**Comment by mattgodbolt** [comment]
Like @partouf says; we rely on the tracking and periodic cleanup we can do with `temp` - there are still some paths in the app that "leak" temporary directories. It's not ideal, of course that should be fixed, but it's something we need else our machines quickly run out of disk space.

**Comment by juanarbol** [comment]
What is the live setup exactly? How can I run it in my machine?

For what I read in my machine, that clean logic was simply broken except for a specific unit test that tracks the temp folders

**Comment by mattgodbolt** [comment]
> What is the live setup exactly? How can I run it in my machine?


`make`

> For what I read in my machine, that clean logic was simply broken except for a specific unit test that tracks the temp folders

I'm not sure what you mean - our cleanup code is definitely working. We periodically check to see if there are any requests in progress, and if not, we tell the `temp` system to clean up any leftover temporary directories (in `lib/handlers/compile.js`.

**Comment by juanarbol** [comment]
> > What is the live setup exactly? How can I run it in my machine?
> 
> 
> 
> 
> 
> `make`
> 
> 
> 
> > For what I read in my machine, that clean logic was simply broken except for a specific unit test that tracks the temp folders
> 
> 
> 
> I'm not sure what you mean - our cleanup code is definitely working. We periodically check to see if there are any requests in progress, and if not, we tell the `temp` system to clean up any leftover temporary directories (in `lib/handlers/compile.js`.

I will take a look 👀

**Comment by juanarbol** [comment]
I'm working on a fix for this, I've checked, and you're entirely right about it.

File: lib/handlers/compile.js [hunk 1]
```diff
@@ -519,6 +511,7 @@ export class CompileHandler {
     }
 }
 
-export function SetTestMode() {
-    hasSetUpAutoClean = true;
+// If the interval is not finished, mocha will hang
+export function ClearIntervalForTestRunner() {
+    clearInterval(cleanInterval);
```
Comment (juanarbol) [inline lib/handlers/compile.js:516]: This is not required; mocha supports a `--exit` flag for this scenario. Please let me know what you think about removing this and running the tests w/ the `--exit` flag.

Learn more at: https://mochajs.org/#-exit

**Comment by partouf** [comment]
Although I appreciate this effort, when looking at the end-result, I prefer to have this in a package dedicated to this task then to semi reproduce it in our code. The calls to the mktempdirs are cleaner in the new way, but it's not worth moving a package into our repo.

**Comment by juanarbol** [comment]
> Although I appreciate this effort, when looking at the end-result, I prefer to have this in a package dedicated to this task then to semi reproduce it in our code. The calls to the mktempdirs are cleaner in the new way, but it's not worth moving a package into our repo and risking bugs.

Yeap, that sounds good to me! Thanks!

**Pull request closed without merging**